tensorboard/plugins/pr_curve/summary.py

```diff
@@ -154,15 +154,24 @@ def op(
         [tp_buckets_v, fp_buckets_v])
     with tf.control_dependencies([initialize_bucket_counts]):
       # Create the non-flat views with class_index as first dimension.
-      tp_buckets = tf.reshape(tp_buckets_v, (num_classes, -1))
-      fp_buckets = tf.reshape(fp_buckets_v, (num_classes, -1))
+      tp_buckets = tf.reshape(tp_buckets_v.read_value(), (num_classes, -1))
+      fp_buckets = tf.reshape(fp_buckets_v.read_value(), (num_classes, -1))
 
       with tf.name_scope('update_op'):
-        # Use scatter_add to update the buckets.
-        update_tp = tf.scatter_add(
-            tp_buckets_v, bucket_indices, true_labels, use_locking=True)
-        update_fp = tf.scatter_add(
-            fp_buckets_v, bucket_indices, false_labels, use_locking=True)
+        # We cannot use tf.scatter_add here because there is no guarantee that
+        # the variable can be read from directly (without the use of the 
+        # read_value method). See
+        # https://github.com/tensorflow/tensorflow/issues/11856 for details.
+        # We hence implement the logic of scatter_add using other functions.
+        new_counts = true_labels + tf.gather(
```

Comment:
Mind calling this `new_true_counts` and likewise `new_false_counts` below? It's slightly confusing to see the same variable used for two decidedly different things.